src/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -273,41 +272,64 @@ public void loadVCFVersion() {
+                    System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
+                            + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
+                            + " automatically adding a corresponding PL field to your VCF header");
+                }
+                addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
             }
+        } else if ( line instanceof VCFFilterHeaderLine ) {
+            final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
+            addMetaDataMapBinding(mFilterMetaData, filterLine.getID(), filterLine);
+        } else if ( line instanceof VCFContigHeaderLine ) {
+            addContigMetaDataLine((VCFContigHeaderLine) line);
+        } else {
+            addMetaDataMapBinding(mOtherMetaData, line.getKey(), line);
         }
+    }
 
-        if ( hasFormatLine(VCFConstants.GENOTYPE_LIKELIHOODS_KEY) && ! hasFormatLine(VCFConstants.GENOTYPE_PL_KEY) ) {
-            if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
-                System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
-                                   + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
-                                   + " automatically adding a corresponding PL field to your VCF header");
+    // add a VCFContigHeaderLine to contigMetaData and avoid adding duplicate contigs
+    private void addContigMetaDataLine(final VCFContigHeaderLine line) {
+        for (VCFContigHeaderLine vcfContigHeaderLine : contigMetaData) {
+            // if we are trying to add a contig for the same ID
+            if (vcfContigHeaderLine.getID().equals(line.getID())) {
+                if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
+                    System.err.println("Found duplicate VCF contig header lines for " + line.getID() + "; keeping the first only" );
+                }
+                // do not add this contig if it exists
+                return;
             }
-            addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
         }
+
+        contigMetaData.add(line);
     }
 
     /**
-     * Add line to map, issuing warnings about duplicates
+     * Add line to map at a given key issuing warnings about duplicates
      *
-     * @param map
-     * @param line
-     * @param <T>
+     * @param map a map from each key to the associated VCFCompoundHeaderLine
+     * @param key the key to insert this line at
+     * @param line the line to insert at this key
+     * @param <T> a type of vcf header line that extends VCFCompoundHeaderLine
```

Comment:
Remove mention of `VCFCompoundHeaderLine` from the docs, since this method now takes anything that extends `VCFHeaderLine` itself.